File: backend/models/documentation.go
package models

import "time"

type Documentation struct {
	ID          int64     `json:"id"`
	Slug        string    `json:"slug"`
	Title       string    `json:"title"`
	Description string    `json:"description"`
	Visibility  string    `json:"visibility"`
	OwnerID     int64     `json:"owner_id"`
	Revision    int64     `json:"revision"`
	CanEdit     bool      `json:"can_edit"`
	CreatedAt   time.Time `json:"created_at"`
	UpdatedAt   time.Time `json:"updated_at"`
}

type DocumentationSection struct {
	ID              int64     `json:"id"`
	DocumentationID int64     `json:"documentation_id"`
	Title           string    `json:"title"`
	DisplayOrder    int       `json:"display_order"`
	CreatedAt       time.Time `json:"created_at"`
	UpdatedAt       time.Time `json:"updated_at"`
}

type DocumentationArticle struct {
	ID              int64     `json:"id"`
	DocumentationID int64     `json:"documentation_id"`
	SectionID       *int64    `json:"section_id,omitempty"`
	Slug            string    `json:"slug"`
	Title           string    `json:"title"`
	Summary         string    `json:"summary"`
	Content         string    `json:"content,omitempty"`
	DisplayOrder    int       `json:"display_order"`
	AuthorID        int64     `json:"author_id"`
	LastEditedBy    int64     `json:"last_edited_by"`
	Revision        int64     `json:"revision"`
	CreatedAt       time.Time `json:"created_at"`
	UpdatedAt       time.Time `json:"updated_at"`
}

type DocumentationManifest struct {
	Documentation *Documentation         `json:"documentation"`
	Sections      []DocumentationSection `json:"sections"`
	Articles      []DocumentationArticle `json:"articles"`
}

type DocumentationArticleView struct {
	Article  *DocumentationArticle `json:"article"`
	Previous *DocumentationArticle `json:"previous,omitempty"`
	Next     *DocumentationArticle `json:"next,omitempty"`
}

type DocumentationSearchResult struct {
	ArticleID int64  `json:"article_id"`
	Slug      string `json:"slug"`
	Title     string `json:"title"`
	Summary   string `json:"summary"`
	Excerpt   string `json:"excerpt"`
	SectionID *int64 `json:"section_id,omitempty"`
}

File: frontend/src/atoms/documentation.ts
export interface Documentation {
  id: number;
  slug: string;
  title: string;
  description: string;
  visibility: "public" | "unlisted" | "private";
  owner_id: number;
  revision: number;
  can_edit: boolean;
  created_at: string;
  updated_at: string;
}

export interface DocumentationSection {
  id: number;
  documentation_id: number;
  title: string;
  display_order: number;
}

export interface DocumentationArticle {
  id: number;
  documentation_id: number;
  section_id?: number;
  slug: string;
  title: string;
  summary: string;
  content?: string;
  display_order: number;
  author_id: number;
  last_edited_by: number;
  revision: number;
  created_at: string;
  updated_at: string;
}

export interface DocumentationManifest {
  documentation: Documentation;
  sections: DocumentationSection[];
  articles: DocumentationArticle[];
}

export interface DocumentationArticleView {
  article: DocumentationArticle;
  previous?: DocumentationArticle;
  next?: DocumentationArticle;
}

export interface DocumentationSearchResult {
  article_id: number;
  slug: string;
  title: string;
  summary: string;
  excerpt: string;
  section_id?: number;
}

export interface ForumDocumentationArticle {
  id: number;
  category_id: number;
  title: string;
  summary: string;
  is_pinned: boolean;
  is_locked: boolean;
  reply_count: number;
  updated_at: string;
}

export interface ForumDocumentationManifest {
  categories: Array<{
    id: number;
    name: string;
    description: string;
    display_order: number;
    is_pinned: boolean;
    is_locked: boolean;
  }>;
  articles: ForumDocumentationArticle[];
}
